library/pkparse.c

```diff
@@ -901,9 +947,14 @@ static int pk_parse_key_sec1_der( mbedtls_ecp_keypair *eck,
             {
                 /*
                  * The only acceptable failure mode of pk_get_ecpubkey() above
-                 * is if the point format is not recognized.
+                 * is if the point format is not recognized. if the underlying
+                 * platform does not support this specific error,
+                 * we should return that specific error.
                  */
-                if( ret != MBEDTLS_ERR_ECP_FEATURE_UNAVAILABLE )
+                if( ret == MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )
```

Comment:
Should this be `(ret & MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED)` if `ret` is a bitfield now?

Reply:
Probably. But I f remember correct, when I tested this, in this specific branch, it was always `MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED`. But better to check the low leverl error

Reply:
I am not sure this is needed. The low level driver could return `MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED`, which in this case it is returned from `pk_get_ecpubkey()`.  In this check, if it is unsupported, we modify the code to `MBEDTLS_ERR_PK_KEY_INVALID_FORMAT | MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED`